packages/client/src/runtime/core/engines/library/LibraryEngine.ts

```diff
@@ -151,11 +180,26 @@ export class LibraryEngine implements Engine<undefined> {
         isolation_level: arg.isolationLevel,
       })
 
-      result = await this.engine?.startTransaction(jsonOptions, headerStr)
+      if (TARGET_BUILD_TYPE === 'rn') {
```

Comment:
This conditionals scattered all over the class look brittle and hard to maintain, let's abstract them out.

We already have an abstraction for using different kind of libraries within `LibraryEngine` and this is `libraryLoader`/ `Library`. What I would suggest:

- We'll write an adapter that would wrap `__PrismaProxy` and make it conform to `Library` interface
- We implement `libraryLoader` for react native. 

If we do that, we can revert most of the `LibraryEngine` to it's previous state and concentrate all the difference in different `Library` implementations. This is how NAPI/WASM library abstraction is implemented and I'd prefer React Native to work the same way too.

Reply:
I've implemented a `reactNativeLibraryLoader`, put the class that wraps the bindings inside, since I didn't know where else to put this. Let me know if this is correct.